src/main/java/com/hms_networks/sc/cumulocity/inventory/InventoryUpdateManager.java

```diff
@@ -100,10 +115,23 @@ public static int[] loadInventoryUpdatePayloads() {
         continue;
       }
 
-      // Update the child device with the file contents
+      // Now that basic checks are done, send update message for the files
       try {
-        updateChildFromFile(files[i]);
-        updateCount[0]++;
+        // With the onlyParentFile flag, only update the parent device
+        if (onlyParentFile) {
+          if (files[i]
+              .getName()
+              .equals(UNIQUE_PARENT_INVENTORY_FILE_NAME + INVENTORY_UPDATE_PAYLOAD_FILE_EXT)) {
+            updateDeviceInventoryObjectFromFile(files[i]);
+            updateCount[0]++;
+          } else {
+            // was not the parent, skip
+            updateCount[1]++;
```

Comment:
I dont see any definition of updateCount that documents the first and second value. I am also not certain why this is an array and not just two ints. Im not sure ive reviewed this file before and its a preexisting construct but i thought i would comment on it since its a bit strange and not clearly documented.

Reply:
`updateCount` is returned and it is described in the javadocs. I wanted to return two values [updates without errors, updates with errors], and did not bother to write a class for this.